compiler_opt/distributed/worker.py

```diff
@@ -28,26 +29,14 @@ def is_priority_method(cls, method_name: str) -> bool:
 
 T = TypeVar('T')
 
-
-# Dask's Futures are limited. This captures that.
-class WorkerFuture(Generic[T], metaclass=abc.ABCMeta):
-
-  @abc.abstractmethod
-  def result(self) -> T:
-    raise NotImplementedError()
-
-  @abc.abstractmethod
-  def done(self) -> bool:
-    raise NotImplementedError()
+WorkerFuture = concurrent.futures.Future
 
 
 def wait_for(futures: Iterable[WorkerFuture]):
   """Dask futures don't support more than result() and done()."""
   for f in futures:
-    try:
-      _ = f.result()
-    except:  # pylint: disable=bare-except
-      pass
+    while not f.done():
```

Comment:
the scheduler returns a dummy future list which slowly gets replaced with real futures, so calling .result is not viable.